### ISSUE TITLE
AddressSanitizer: a memory leak of dinf_New()

### ISSUE THREAD

**Issue by Clingto** [gpac/gpac#1321]
System info:  
Ubuntu 16.04.6 LTS, X64, gcc 5.4.0, gpac (master 6ada10e)  
Compile Command:
```
$ CC="gcc -fsanitize=address -g" CXX="g++ -fsanitize=address -g" ./configure --static-mp4box
$ make
```
Run Command:
```
$ MP4Box -diso -out /dev/null $POC-memory-leak
```
POC file:  
https://github.com/Clingto/POC/blob/master/gpac-MP4Box/POC-memory-leak  
ASAN info:
```C
==26273==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 80 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff6f02602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0xad2eed in dinf_New isomedia/box_code_base.c:945

Indirect leak of 160 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff6f02961 in realloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98961)
    #1 0x46c43a in realloc_chain utils/list.c:622
    #2 0x46c43a in gf_list_add utils/list.c:629

Indirect leak of 96 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff6f02602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0xad1d0d in url_New isomedia/box_code_base.c:575

Indirect leak of 80 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff6f02602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0xad318d in dref_New isomedia/box_code_base.c:1005

Indirect leak of 32 byte(s) in 2 object(s) allocated from:
    #0 0x7ffff6f02602 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x98602)
    #1 0x46c27d in gf_list_new utils/list.c:602

SUMMARY: AddressSanitizer: 448 byte(s) leaked in 10 allocation(s).

```
---
Edit

This bug issue still exists in latest version 0.8.0: 4c19ae5 and 0.9.0: 1de1f8d

---

Addition: This bug was found with our fuzzer, which is based on AFL. Our fuzzer is developed by Yuanpingyu(cfenicey@gmail.com) 、Yanhao and Marsman1996(lqliuyuwei@outlook.com)

**Comment by carnil** [comment]
CVE-2019-20159 was assigned for this issue.

**Comment by aureliendavid** [comment]
Thanks for the report. 

This should be fixed in master / 0.8.0 as of the above commit. 

It will be included in filters / 0.9.0 in the next merge. 

Feel free to reopen the issue if necessary.
